spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java

```diff
@@ -0,0 +1,263 @@
+
+	@Override
+	public void purgeQueue(String vhost, Queue queue) {
+		this.rabbitClient.purgeQueue(vhost, queue.getName());
+	}
+
+	@Override
+	public void deleteQueue(Queue queue) {
+		this.rabbitClient.deleteQueue(DEFAULT_VHOST, queue.getName());
+	}
+
+	@Override
+	public void deleteQueue(String vhost, Queue queue) {
+		this.rabbitClient.deleteQueue(vhost, queue.getName());
+	}
+
+	@Override
+	public Queue getQueue(String name) {
+		return getQueue(DEFAULT_VHOST, name);
+	}
+
+	@Override
+	public Queue getQueue(String vhost, String name) {
+		return convert(this.rabbitClient.getQueue(vhost, name));
+	}
+
+	@Override
+	public List<Queue> getQueues() {
+		return converteQueueList(this.rabbitClient.getQueues());
+	}
+
+	@Override
+	public List<Queue> getQueues(String vhost) {
+		return converteQueueList(this.rabbitClient.getQueues(vhost));
+	}
+
+	@Override
+	public void addQueue(Queue queue) {
+		addQueue(DEFAULT_VHOST, queue);
+	}
+
+	@Override
+	public void addQueue(String vhost, Queue queue) {
+		QueueInfo info = new QueueInfo();
+		info.setArguments(queue.getArguments());
+		info.setAutoDelete(queue.isAutoDelete());
+		info.setDurable(queue.isDurable());
+		info.setExclusive(queue.isExclusive());
+		this.rabbitClient.declareQueue(vhost, queue.getName(), info);
+	}
+
+	@Override
+	public void deleteExchange(Exchange exchange) {
+		this.rabbitClient.deleteQueue(DEFAULT_VHOST, exchange.getName());
+	}
+
+	@Override
+	public void deleteExchange(String vhost, Exchange exchange) {
+		this.rabbitClient.deleteExchange(vhost, exchange.getName());
+	}
+
+	@Override
+	public Exchange getExchange(String name) {
+		return getExchange(DEFAULT_VHOST, name);
+	}
+
+	@Override
+	public Exchange getExchange(String vhost, String name) {
+		return convert(this.rabbitClient.getExchange(vhost, name));
+	}
+
+	@Override
+	public List<Exchange> getExchanges() {
+		return converteExchangeList(this.rabbitClient.getExchanges());
+	}
+
+	@Override
+	public List<Exchange> getExchanges(String vhost) {
+		return converteExchangeList(this.rabbitClient.getExchanges(vhost));
+	}
+
+	@Override
+	public List<Binding> getBindings() {
+		return converteBindingList(this.rabbitClient.getBindings());
+	}
+
+	@Override
+	public List<Binding> getBindings(String vhost) {
+		return converteBindingList(this.rabbitClient.getBindings(vhost));
+	}
+
+	@Override
+	public List<Binding> getBindingsForExchange(String vhost, String exchange) {
+		return converteBindingList(this.rabbitClient.getBindingsBySource(vhost, exchange));
+	}
+
+	private List<Queue> converteQueueList(List<QueueInfo> queues) {
+		List<Queue> convertedQueues = new ArrayList<Queue>();
+		for (QueueInfo qi : queues) {
+			convertedQueues.add(convert(qi));
+		}
+		return convertedQueues;
+	}
+
+	private Queue convert(QueueInfo qi) {
+		return new Queue(qi.getName(), qi.isDurable(), qi.isExclusive(), qi.isAutoDelete(),
+				qi.getArguments());
+	}
+
+	private List<Exchange> converteExchangeList(List<ExchangeInfo> exchanges) {
```

Comment:
**converte** ? It is valid word, buy the way: http://en.wikipedia.org/wiki/Galician_language
But I hope we prefer English :smile: 
Will fix on merge.